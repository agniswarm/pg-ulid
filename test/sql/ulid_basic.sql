-- Test basic ULID functionality
CREATE EXTENSION IF NOT EXISTS ulid;

-- Test ulid() function
SELECT ulid() IS NOT NULL AS ulid_generated;

-- Test ulid_random() function
SELECT ulid_random() IS NOT NULL AS ulid_random_generated;

-- Test ulid_time() function
SELECT ulid_time((extract(epoch from now()) * 1000)::BIGINT) IS NOT NULL AS ulid_time_generated;

-- Test ulid_batch() function
SELECT array_length(ulid_batch(5), 1) = 5 AS ulid_batch_test;

-- Test ulid_random_batch() function
SELECT array_length(ulid_random_batch(3), 1) = 3 AS ulid_random_batch_test;

-- Test ulid_parse() function
SELECT * FROM ulid_parse('01K4FQ7QN4ZSW0SG5XACGM2HB4') IS NOT NULL AS ulid_parse_test;
