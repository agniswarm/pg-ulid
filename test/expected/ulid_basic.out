-- Test basic ULID functionality
CREATE EXTENSION IF NOT EXISTS ulid;

-- Test ulid() function
SELECT ulid() IS NOT NULL AS ulid_generated;
 ulid_generated 
----------------
 t
(1 row)

-- Test ulid_random() function
SELECT ulid_random() IS NOT NULL AS ulid_random_generated;
 ulid_random_generated 
-----------------------
 t
(1 row)

-- Test ulid_time() function
SELECT ulid_time((extract(epoch from now()) * 1000)::BIGINT) IS NOT NULL AS ulid_time_generated;
 ulid_time_generated 
---------------------
 t
(1 row)

-- Test ulid_batch() function
SELECT array_length(ulid_batch(5), 1) = 5 AS ulid_batch_test;
 ulid_batch_test 
-----------------
 t
(1 row)

-- Test ulid_random_batch() function
SELECT array_length(ulid_random_batch(3), 1) = 3 AS ulid_random_batch_test;
 ulid_random_batch_test 
------------------------
 t
(1 row)

-- Test ulid_parse() function
SELECT * FROM ulid_parse('01K4FQ7QN4ZSW0SG5XACGM2HB4') IS NOT NULL AS ulid_parse_test;
 ulid_parse_test 
-----------------
 t
(1 row)
